batch/batch/cloud/gcp/driver/create_instance.py

```diff
@@ -44,7 +44,7 @@ def create_vm_config(
 
     region = instance_config.region_for(zone)
 
-    if local_ssd_data_disk:
+    if local_ssd_data_disk or 'g2' in machine_type:
```

Comment:
I think this logic is not quite right.  A pd-ssd disk attached to a g2 shows up as an NVMe drive, but as SCSI drive on a n1/n2.  But if local_ssd is not set, we should not switch the disk type to local-ssd.  I think you can check the machine type in the false branch to set the disk name.